client/dive-common/use/useSave.ts

```diff
@@ -89,59 +104,84 @@ export default function useSave(
-      if (action === 'meta') {
-        pendingChangeMap.meta += 1;
-      } else if (track !== undefined) {
-        _updatePendingChangeMap(
-          track.trackId,
-          track,
-          action,
-          pendingChangeMap.upsert,
-          pendingChangeMap.delete,
-        );
-      } else if (attribute !== undefined) {
-        _updatePendingChangeMap(
-          attribute.key,
-          attribute,
-          action,
-          pendingChangeMap.attributeUpsert,
-          pendingChangeMap.attributeDelete,
-        );
-      } else if (group !== undefined) {
-        _updatePendingChangeMap(
-          group.id,
-          group,
-          action,
-          pendingChangeMap.groupUpset,
-          pendingChangeMap.groupDelete,
-        );
-      } else {
-        throw new Error(`Arguments inconsistent with pending change type: ${action} cannot be performed without additional arguments`);
+    if (pendingChangeMaps[cameraName]) {
+      const pendingChangeMap = pendingChangeMaps[cameraName];
+
+      if (!readonlyMode.value) {
+        if (action === 'meta') {
+          pendingChangeMap.meta += 1;
+        } else if (track !== undefined) {
+          _updatePendingChangeMap(
+            track.trackId,
+            track,
+            action,
+            pendingChangeMap.upsert,
+            pendingChangeMap.delete,
+          );
+        } else if (attribute !== undefined) {
+          _updatePendingChangeMap(
+            attribute.key,
+            attribute,
+            action,
+            pendingChangeMap.attributeUpsert,
+            pendingChangeMap.attributeDelete,
+          );
+        } else if (group !== undefined) {
+          _updatePendingChangeMap(
+            group.id,
+            group,
+            action,
+            pendingChangeMap.groupUpset,
+            pendingChangeMap.groupDelete,
+          );
+        } else {
+          throw new Error(`Arguments inconsistent with pending change type: ${action} cannot be performed without additional arguments`);
+        }
+        pendingSaveCount.value += 1;
       }
-      pendingSaveCount.value += 1;
     }
   }
 
   function discardChanges() {
-    pendingChangeMap.upsert.clear();
-    pendingChangeMap.delete.clear();
-    pendingChangeMap.attributeUpsert.clear();
-    pendingChangeMap.attributeDelete.clear();
-    pendingChangeMap.groupUpset.clear();
-    pendingChangeMap.groupDelete.clear();
-    pendingChangeMap.meta = 0;
+    Object.values(pendingChangeMaps).forEach((pendingChangeMap) => {
+      pendingChangeMap.upsert.clear();
+      pendingChangeMap.delete.clear();
+      pendingChangeMap.attributeUpsert.clear();
+      pendingChangeMap.attributeDelete.clear();
+      pendingChangeMap.groupUpset.clear();
+      pendingChangeMap.groupDelete.clear();
+      // eslint-disable-next-line no-param-reassign
+      pendingChangeMap.meta = 0;
+    });
     pendingSaveCount.value = 0;
   }
 
+  function addCamera(cameraName: string) {
```

Comment:
I find it slightly confusing that `CameraStore` has a `removeCamera` method but this file does not.